p2p/src/network/pubsub/p2p_network_pubsub_effects.rs

```diff
@@ -111,16 +113,55 @@ impl P2pNetworkPubsubAction {
                     topic: TOPIC.to_owned(),
                 });
             }
-            P2pNetworkPubsubAction::Sign { .. } => {
+            P2pNetworkPubsubAction::Sign { author, topic, .. } => {
                 if let Some(to_sign) = state.to_sign.front() {
                     let mut publication = vec![];
-                    prost::Message::encode(to_sign, &mut publication).unwrap();
-                    let signature = store.service().sign_publication(&publication).into();
-                    store.dispatch(P2pNetworkPubsubAction::BroadcastSigned { signature });
+                    if prost::Message::encode(to_sign, &mut publication).is_err() {
+                        store.dispatch(P2pNetworkPubsubAction::SignError { author, topic });
+                    } else {
+                        let signature = store.service().sign_publication(&publication).into();
+                        store.dispatch(P2pNetworkPubsubAction::BroadcastSigned { signature });
+                    }
                 }
             }
             P2pNetworkPubsubAction::BroadcastSigned { .. } => broadcast(store),
-            P2pNetworkPubsubAction::IncomingData { peer_id, .. } => {
+            P2pNetworkPubsubAction::IncomingData {
+                peer_id,
+                seen_limit,
+                ..
+            } => {
+                let Some(state) = state.clients.get(&peer_id) else {
+                    return;
```

Comment:
Is this an error condition or something normal?

Reply:
if the enabling condition is correct, this should never fail because the existence of such peer_id in the collection should have been checked already.

But still in general, as part of the refactor (in a later phase) reducers should return a `Result` and all these cases should return an error that we can log so that this never fails silently if we have a bug

Reply:
Maybe we could add later `bug_condition!()` in such cases, that doesn't require immediate changing the signatures of reducers and propagate the Result. Also I could include in the error message for `bug_condition!()` that the possible cause is that the enabling condition is not strong enough.